### ISSUE TITLE
[BUG]: DateTimeOffsetConverter doesn't parse null completed_at

### ISSUE THREAD

**Issue by itrion** [octokit/webhooks.net#595]
### What happened?

Similartly to https://github.com/octokit/webhooks.net/issues/140, the DateTimeOffsetConverter is throwing 
```
System.Text.Json.JsonException
'Null' is not an allowed token type for the DateTimeOffsetConverter
   at Octokit.Webhooks.Converter.DateTimeOffsetConverter.ReadInternal(Utf8JsonReader& reader)
   at Octokit.Webhooks.Converter.DateTimeOffsetConverter.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options)
...
```
when parsing a `workflow_job` event with several `"completed_at": null`. The completed_at with null values are in
- `.workflow_job.completed_at`
- and `.workflow_job.steps[N].completed_at` 

### Versions

Octokit.Webhooks.AspNetCore 2.4.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JamieMagee** [comment]
@itrion do you have a sample payload that I can test with? It looks like `workflow_job.completed_at` is already nullable (though it's `string?` instead of `DateTimeOffset?`

https://github.com/octokit/webhooks.net/blob/ecab931d8a1319de2d9da5b31d8668c004ad2df0/src/Octokit.Webhooks/Models/WorkflowJobEvent/WorkflowJob.cs#L66-L67

And `workflow_job.steps[].completed_at` is already `DateTimeOffset?` and uses the [`NullableDateTimeOffsetConverter`][1]

https://github.com/octokit/webhooks.net/blob/ecab931d8a1319de2d9da5b31d8668c004ad2df0/src/Octokit.Webhooks/Models/WorkflowJobEvent/WorkflowJobStep.cs#L24-L26

[1]: https://github.com/octokit/webhooks.net/blob/main/src/Octokit.Webhooks/Converter/NullableDateTimeOffsetConverter.cs

**Comment by itrion** [comment]
Thank you @JamieMagee, after a deeper troubleshooting I realized it was actually failing parsing the `started_at`. Let me know if you need a full payload or if this is enough

![{BA102EA2-73F3-4BBE-B129-7F6EF2668CCB}](https://github.com/user-attachments/assets/ed8dbd66-dabb-4e0f-9bef-79f4fd7dd465)

```json
{
	"action": "in_progress",
	"workflow_job": {
		"id": 123,
		"run_id": 1234,
		"workflow_name": "docker in  - Update",
		"head_branch": "main",
		"run_url": "https://api.github.com/repos/example/example/actions/runs/11439205892",
		"run_attempt": 1,
		"node_id": "CR_123",
		"head_sha": "96bd5bd7ac9cbc35a15cea8262abbd385778342b",
		"url": "https://api.github.com/repos/example/example/actions/jobs/31822394947",
		"html_url": "https://github.com/example/example/actions/runs/11439205892/job/31822394947",
		"status": "in_progress",
		"conclusion": null,
		"created_at": "2024-10-21T11:32:36Z",
		"started_at": "2024-10-21T11:32:47Z",
		"completed_at": null,
		"name": "Dependabot",
		"steps": [
			{
				"name": "Set up job",
				"status": "completed",
				"conclusion": "success",
				"number": 1,
				"started_at": "2024-10-21T11:32:47Z",
				"completed_at": "2024-10-21T11:32:48Z"
			},
			{
				"name": "Verify actions runner dependencies",
				"status": "in_progress",
				"conclusion": null,
				"number": 2,
				"started_at": "2024-10-21T11:32:48Z",
				"completed_at": null
			},
			{
				"name": "Create job directory",
				"status": "pending",
				"conclusion": null,
				"number": 3,
				"started_at": null,
				"completed_at": null
			},
			{
				"name": "Run Dependabot",
				"status": "pending",
				"conclusion": null,
				"number": 4,
				"started_at": null,
				"completed_at": null
			},
			{
				"name": "Cleanup job directory",
				"status": "pending",
				"conclusion": null,
				"number": 5,
				"started_at": null,
				"completed_at": null
			}
		],
		"check_run_url": "https://api.github.com/repos/example/example/check-runs/31822394947",
		"labels": [
			"dependabot"
		],
		"runner_id": 123,
		"runner_name": "dependabot",
		"runner_group_id": 25,
		"runner_group_name": "1234"
	}
}
```

I got to reproduce it with that payload running
```csharp
	[Fact]
	public async Task ReproduceError()
	{
		var payload = await File.ReadAllTextAsync($"Fixtures/test-github-webhook-workflow_job-null-issue.json");
		var workflowJobEvent = JsonSerializer.Deserialize<WorkflowJobEvent>(payload)!;
		workflowJobEvent.Should().NotBeNull();
	}
```

**Comment by JamieMagee** [comment]
Thank you! This should be resolved by #596
